x-pack/plugins/encrypted_saved_objects/server/saved_objects/encrypted_saved_objects_client_wrapper.ts

```diff
@@ -323,8 +324,8 @@ function getValidId(
     // and not being used to get around the specified ID limitation
     const canSpecifyID = (overwrite && version) || SavedObjectsUtils.isRandomId(id);
     if (!canSpecifyID) {
-      throw new Error(
-        'Predefined IDs are not allowed for saved objects with encrypted attributes, unless the ID has been generated using `SavedObjectsUtils.generateId`.'
+      throw Boom.badRequest(
+        'Predefined IDs are not allowed for saved objects with encrypted attributes unless the ID is a UUID.'
```

Comment:
@elastic/kibana-security, I thought of making the error message more generic when it is returned by an HTTP call (ex: when creating an alert). Let me know your thoughts?

Reply:
I'm fine with the text change. Is this something you expect end users to organically encounter in the UI?

Reply:
@legrego 

>Is this something you expect end users to organically encounter in the UI?

I don't think UI but I can see 3rd party developers/users encountering this when calling our HTTP API. I wasn't sure if mentioning `SavedObjectsUtils.generateId` in the error message would make sense to them or if it was better to say any UUID? I'm good either way.

Reply:
I think your text change makes sense, and I have no problems with API consumers seeing this updated message. I just wanted to make sure the "normal" UI-based experience wouldn't encounter this 🙂

Reply:
Awesome, good to know 🙂 I can't guarantee what solutions will build on top, but I agree this shouldn't be encountered in the UIs.